docs: update UG (#162)
AY2122S2-CS2103T-T17-2/tp: docs/UserGuide.md

    3. Toss the spaghetti in the sauce,
    4. Taste and season with salt and black pepper.

 [<sub>return to table of content<sub>](#table-of-content)

### Edit an existing recipe: `edit`
Edits a recipe by specifying its name, ingredients, and steps to prepare the dish.<br>

> Usage: `edit -n <name of recipe> -d <completion time of recipe> -ss <serving size of recipe> -i <ingredients of recipes separated by commas> -s <steps separated by commas> -t <tags associated with recipe> (optional)` <br>

Author: Actually for this part (and the command summary) do you wanna do the update? Since it's your feature HAAHHA. As discussed earlier, we should probably standardise index to number for UG.

    3. Toss the spaghetti in the sauce,
    4. Taste and season with salt and black pepper.

 [<sub>return to table of content<sub>](#table-of-content)

### Edit an existing recipe: `edit`
Edits a recipe by specifying its name, ingredients, and steps to prepare the dish.<br>

<div markdown="block" class="alert alert-info">

:information_source: **Only specify the parts of the recipe that you want to edit!** The only necessary information needed is the name or number of the recipe that you would like to edit<br>

</div>

> Usage: `edit <name of recipe to be edited> [-n <new name of recipe>] [-d <new completion time of recipe>] [-ss <new serving size of recipe>] [-i <new ingredients of recipes separated by commas>] [-s <new steps separated by commas>] [-t <new tags associated with recipe>]` <br>

Reviewer: ```suggestion
Usage: `edit <name of recipe to be edited> [-n <new name of recipe> -d <new completion time of recipe> -ss <new serving size of recipe> -i <new ingredients of recipes separated by commas> -s <new steps separated by commas> -t <new tags associated with recipe>]` <br>
```

   OR <br>

> `edit -x <index of recipe to be edited> -n <new name of recipe> -d <new completion time of recipe> -ss <new serving size of recipe> -i <new ingredients of recipes separated by commas> -s <new steps separated by commas> [-t <new tags associated with recipe>]`

Reviewer: ```suggestion
`edit -x <number of recipe to be edited> [-n <new name of recipe> -d <new completion time of recipe> -ss <new serving size of recipe> -i <new ingredients of recipes separated by commas> -s <new steps separated by commas> -t <new tags associated with recipe>]`
```

Example:
- `edit Aglio Olio -n Spicy Aglio Olio -i spaghetti 56 grams, garlic 3 cloves, bacon 3 slices, olive oil 110 ml, Chilli flakes 10 teaspoons -s Cook the spaghetti until al dante, Saute the chopped garlic, Toss spaghetti in the sauce, taste and season with salt, black pepper and chilli flakes`

>     - This edits the recipe named "Aglio Olio" and updates its attributes with a name of "Spicy Aglio Olio", completion time of 5 mins, serving size of 1 person/pax and ingredients consisting of 56 grams of spaghetti, 3 cloves of garlic, 3 slices of bacon, 110 ml of olive oil and 10 teaspoons of chilli flakes with the steps of:

Reviewer: Example edit command doesn't update the serving size or completion time - p sure was my mistake i rmb typing this

```suggestion
    - This edits the recipe named "Aglio Olio" and updates its attributes with a name of "Spicy Aglio Olio" and ingredients consisting of 56 grams of spaghetti, 3 cloves of garlic, 3 slices of bacon, 110 ml of olive oil and 10 teaspoons of chilli flakes with the steps of:
```

</div>

[<sub>return to table of content<sub>](#table-of-content)

### Clear (delete all) existing recipes: `clear`
Clears the entire recipe book. A confirmation prompt would show up before the user could clear immediately. <br>

Usage: `clear`

The program would then prompt the user to confirm. Type in `yes` to clear and `no` to cancel the clear request.

Example: 
- clear
    - This would generate a clear request that requires a confirmation. To confirm to clear the recipe book, type in 'yes',
    - else, type in 'no'.

<div markdown="block" class="alert alert-info">

️:heavy_exclamation_mark: :computer: To clear the recipe book without the confirmation, users simply have to include the prefix `-f` after the `clear` command. McKitchen is only able to restore the default sample recipes by running `reset`. Currently, McKitchen isn't able to undo actions in the application so use this feature with caution! 

</div>

#### Forced Clear:
Usage: `clear -f` <br>
Example:
- clear -f
    - This would clear the recipe book without having the need for any confirmation.

[<sub>return to table of content<sub>](#table-of-content)

### Save recipes

> Save the recipes in a human-readable text file on the your computer and is modifiable through a text editor if the user is familiar with the format (JSON). <br>

Reviewer: ```suggestion
Save the recipes in a human-readable text file on your computer and is modifiable through a text editor if the user is familiar with the format (JSON). <br>
```

Usage: (Automatically updates text file upon each modifying (writing) operation).

<div markdown="block" class="alert alert-info">

️:information_source: Recipes would be saved in a folder called "data" from where McKitchen is located. <br>
Example: If McKitchen is saved in the "Downloads" folder, recipes would be saved in a folder called "data" in "Downloads" as "recipebook.json".

️:computer: If you are familiar with how file paths works in your operating system, the "recipebook.json" file would be created in a folder **relative** to McKitchen.jar (data/recipebook.json) 

</div>

[<sub>return to table of content<sub>](#table-of-content)

### Load recipe
Load existing recipes from the text file on the user's computer. Modifications to this file would be displayed accordingly as long as the format is followed. <br>
Usage: (Automatically loads the recipes upon launching the application).

<div markdown="block" class="alert alert-info">

️:information_source: Recipes would be loaded from a file called "recipebook.json" in a folder called "data" from where McKitchen is located. <br>

> Example: If McKitchen is stored in the "Downloads" folder, recipes would be loaded in a folder called "data" which should "recipebook.json". <br><br>

Reviewer: ```suggestion
Example: If McKitchen is stored in the "Downloads" folder, recipes would be loaded in a folder called "data" which should contain "recipebook.json". <br><br>
```

> If the file or folder does not exist, McKitchen would automatically create them and provide a sample recipes to allow users to explore the application without needing to input their own recipes. <br>

Reviewer: ```suggestion
If the file or folder does not exist, McKitchen would automatically create them and provide sample recipes to allow users to explore the application without needing to input their own recipes. <br>
```

        3. Toss the spaghetti in the sauce,
        4. Taste and season with salt, black pepper and chilli flakes.

[<sub>return to table of content<sub>](#table-of-content)

### Delete an existing recipe: `delete`

> Removes a stored recipe based on the number it is associated with in the `list` <br>

Reviewer: ```suggestion
Removes a stored recipe based on the index it is associated with in the `list` <br>
```

Usage: `delete -x <recipe index>` or `delete <recipe name>` <br> 
Example:
- `delete -x 1`
    - This would delete the first entry in the list of recipes.
- `delete Aglio Olio` 
    - This would delete the recipe called 'Aglio Olio'.

<div markdown="block" class="alert alert-info">

:information_source: **Deletion of recipes by its name are case-sensitive!**<br>
To prevent accidental deletions, McKitchen makes it _slightly_ inconvenient for users to delete a recipe by its name by making it case-sensitive. A faster way to remove a recipe from McKitchen is through its index.

> </div>

Author: I'll update it to case-insensitive and say that we did it for their convenience instead HAHAHA

[<sub>return to table of content<sub>](#table-of-content)

> ### Find recipe an existing recipe: `find`

Reviewer 2: ```suggestion
### Find an existing recipe: `find`
```

Searches for a recipe based on the keywords (seperated by comma). <br>
Usage: `find <keyword> [other keywords, seperated by commas]` <br>

Example: 
- Let's say that you would like to cook a Western cuisine and have spaghetti and garlic on hand. <br>
        The command that you would execute should look something like this:
    - `find western, spaghetti, garlic`

<div markdown="block" class="alert alert-info">

:information_source: **Keywords used for the find feature are case-insensitive!**<br>
Don't worry, McKitchen would still be able to find "Secret Family Sauce" even if you search for "secret family sauce" (or searching for the secret ingredient) :wink:

</div>

[<sub>return to table of content<sub>](#table-of-content)

### List all stored recipes: `list`
Displays all the available recipes that has been stored locally <br>
Usage: `list`

[<sub>return to table of content<sub>](#table-of-content)

> ### View recipe an existing recipe: `view`

Reviewer 2: Should be "View an exisiting recipe" instead of "View recipe an existing recipe" to be consistent with Delete
```suggestion
### View an existing recipe: `view`
```

Author: omg how did i miss those HAHAHA, thanks for catching them!

View the contents of an existing stored recipe based on recipe name or index. <br>

> Usage: `view <recipe name>` or `view -x <recipe number from list>` <br>

Reviewer: ```suggestion
Usage: `view -x <recipe index from list>` or `view <recipe name>` <br>
```

Example:
- `view Aglio Olio`
    - This would display the full contents of the recipe matching the name "Aglio Olio" including its ingredients and steps to prepare the dish.

- `view -x 1`
    - This would display the full contents of the first recipe in the list which includes its ingredients and steps to prepare the dish.

<div markdown="block" class="alert alert-info">

> ️:information_source: Recipe names are case-sensitive!

Reviewer: p sure its case insensitive nowww

Example: 
- clear
    - This would generate a clear request that requires a confirmation. To confirm to clear the recipe book, type in 'yes',
    - else, type in 'no'.

<div markdown="block" class="alert alert-info">

️:heavy_exclamation_mark: :computer: To clear the recipe book without the confirmation, users simply have to include the prefix `-f` after the `clear` command. McKitchen is only able to restore the default sample recipes by running `reset`. Currently, McKitchen isn't able to undo actions in the application so use this feature with caution! 

</div>

#### Forced Clear:
Usage: `clear -f` <br>
Example:
- clear -f
    - This would clear the recipe book without having the need for any confirmation.

[<sub>return to table of content<sub>](#table-of-content)

### Save recipes
Save the recipes in a human-readable text file on the your computer and is modifiable through a text editor if the user is familiar with the format (JSON). <br>
Usage: (Automatically updates text file upon each modifying (writing) operation).

<div markdown="block" class="alert alert-info">

️:information_source: Recipes would be saved in a folder called "data" from where McKitchen is located. <br>

> Example: If McKitchen is saved in the "Downloads" folder, recipes would be saved in a folder called "data" in "Downloads" as "recipebook.json".

Reviewer: perhaps add the file paths to make it easier for dev readers to understand? e.g. /downloads, /download/data/recipebook.json

</div>

[<sub>return to table of content<sub>](#table-of-content)

### Load recipe
Load existing recipes from the text file on the user's computer. Modifications to this file would be displayed accordingly as long as the format is followed. <br>
Usage: (Automatically loads the recipes upon launching the application).

<div markdown="block" class="alert alert-info">

️:information_source: Recipes would be loaded from a file called "recipebook.json" in a folder called "data" from where McKitchen is located. <br>

> Example: If McKitchen is stored in the "Downloads" folder, recipes would be loaded in a folder called "data" which should "recipebook.json". <br><br>

Reviewer: perhaps add the file paths to make it easier for dev readers to understand? e.g. /downloads, /download/data/recipebook.json

Author: For this part I just added the :computer: which I think is a good comprpomise

If the file or folder does not exist, McKitchen would automatically create them and provide a sample recipes to allow users to explore the application without needing to input their own recipes.
</div>

<div markdown="block" class="alert alert-info">

> ️:computer: Advance user tip: If you are familiar with JSON, you modify the file to add new recipes or modify aspects of an existing recipe without needing to run McKitchen. You are advised to strictly follow the formatting of the application as failure to do so may result to certain aspects of the recipes to not load properly, if at all.

Reviewer: ```suggestion
️:computer: Advance user tip: If you are familiar with JSON, you can modify the file to add new recipes or modify aspects of an existing recipe without needing to run McKitchen. You are advised to strictly follow the formatting of the application as failure to do so may result to certain aspects of the recipes to not load properly, if at all.
```

    3. Toss the spaghetti in the sauce,
    4. Taste and season with salt and black pepper.

 [<sub>return to table of content<sub>](#table-of-content)

### Edit an existing recipe: `edit`
Edits a recipe by specifying its name, ingredients, and steps to prepare the dish.<br>

<div markdown="block" class="alert alert-info">

:information_source: **Only specify the parts of the recipe that you want to edit!** The only necessary information needed is the name or number of the recipe that you would like to edit<br>

</div>

Usage: `edit <name of recipe to be edited> [-n <new name of recipe> -d <new completion time of recipe> -ss <new serving size of recipe> -i <new ingredients of recipes separated by commas> -s <new steps separated by commas> -t <new tags associated with recipe>]` <br>
   OR <br>

> `edit -x <recipe number to be edited> -n <new name of recipe> -d <new completion time of recipe> -ss <new serving size of recipe> -i <new ingredients of recipes separated by commas> -s <new steps separated by commas> [-t <new tags associated with recipe>]`

Reviewer: ```suggestion
`edit -x <recipe number to be edited> [-n <new name of recipe> -d <new completion time of recipe> -ss <new serving size of recipe> -i <new ingredients of recipes separated by commas> -s <new steps separated by commas> -t <new tags associated with recipe>]`
```

Example:
- `edit Aglio Olio -n Spicy Aglio Olio -i spaghetti 56 grams, garlic 3 cloves, bacon 3 slices, olive oil 110 ml, Chilli flakes 10 teaspoons -s Cook the spaghetti until al dante, Saute the chopped garlic, Toss spaghetti in the sauce, taste and season with salt, black pepper and chilli flakes`
    - This edits the recipe named "Aglio Olio" and updates its attributes with a name of "Spicy Aglio Olio" and ingredients consisting of 56 grams of spaghetti, 3 cloves of garlic, 3 slices of bacon, 110 ml of olive oil and 10 teaspoons of chilli flakes with the steps of:
        1. Cook the spaghetti until al dante,
        2. Saute the chopped garlic,
        3. Toss the spaghetti in the sauce,
        4. Taste and season with salt, black pepper and chilli flakes.

- `edit -x 1 -n Spicy Aglio Olio [and other information that you would like to edit]`

[<sub>return to table of content<sub>](#table-of-content)

### Delete an existing recipe: `delete`
Removes a stored recipe based on the number it is associated with in the `list` <br>

> Usage: `delete <recipe name>` or `delete -x <recipe number>` <br> 

Reviewer: ```suggestion
Usage: `delete <recipe name>` or `delete -x <recipe number from list>` <br> 
```

You cannot save the recipe with empty fields. Instead, you can add a dummy value into the field you are unsure of, then edit it after you are sure of the input values.

**Q: Is there a faster way to delete a recipe instead of copying the recipe name word for word?**<br>
The fastest way is to delete a recipe by its index `delete -x <recipe index>`

**Q: How do I transfer my recipes to another computer?**<br>
Copy the recipebook.json file into your new computer.

**Q: What do I need to consider when adding a new recipe?**<br>

> You need to consider the recipe name, ingredients, steps to prepare the dish, and it's completion time. You can also consider adding tags associated with your recipe.

Reviewer: missing serving size :(

Reviewer 3: What size am I?

   2. **Optional** details of the command that are enclosed in square brackets such as [optional details].
3. What does it mean when the input command says "... separated by commas"?

| Action               | Command format                                                                                                                                                                                                 |
|----------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| add a new recipe     | `add -n <name of recipe> -d <prep time of recipe> -ss <serving size/portions of recipe> -i <ingredients separated by commas> -s <steps separated by commas> [-t <tags separated by commas>]`                   |
| edit a recipe        | `edit <name of recipe> -n <name of recipe> -d <prep time of recipe> -ss <serving size/portions of recipe> -i <ingredients separated by commas> -s <steps separated by commas> [-t <tags separated by commas>]` |

> | delete a recipe      | `delete -x <recipe number from list>` <br> or <br> `delete <recipe name>`                                                                                                                                      |

Reviewer: ```suggestion
| delete a recipe      | `delete -x <recipe index from list>` <br> or <br> `delete <recipe name>`                                                                                                                                      |
```

Reviewer 3: 🔥

| find a recipe        | `find <keyword> [other keywords separated by commas]`                                                                                                                                                          |
| list a recipe        | `list`                                                                                                                                                                                                         |

> | view specific recipe | `view <recipe number from list>` <br/> or <br/> `view <recipe name>`                                                                                                                                           |

Reviewer: ```suggestion
| view specific recipe | `view <recipe index from list>` <br/> or <br/> `view <recipe name>`                                                                                                                                           |
```

Reviewer 3: 🔥Aidan so sharp

> | view specific recipe | `view <recipe number from list>` <br/> or <br/> `view <recipe name>`                                                                                                                                           |

Reviewer 2: Should change this to make it consistent with delete command
```suggestion
| view specific recipe | `view -x <recipe index from list>` <br/> or <br/> `view <recipe name>`                                                                                                                                           |
```

Author: went to edit the delete command instead since the command explanation follows the name-then-index slow

1. User actions (what you would want to do) is on the left while the command associated to it is on the right.
2. Almost all the details of the commands are needed to be given before the command can be executed properly.
   1. **Compulsory** details are denoted by angled-brackets such as \<mandatory details>.
   2. **Optional** details of the command that are enclosed in square brackets such as [optional details].
3. What does it mean when the input command says "... separated by commas"?

| Action               | Command format                                                                                                                                                                                                 |
|----------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| add a new recipe     | `add -n <name of recipe> -d <prep time of recipe> -ss <serving size/portions of recipe> -i <ingredients separated by commas> -s <steps separated by commas> [-t <tags separated by commas>]`                   |

> | edit a recipe        | `edit <name of recipe> -n <name of recipe> -d <prep time of recipe> -ss <serving size/portions of recipe> -i <ingredients separated by commas> -s <steps separated by commas> [-t <tags separated by commas>]` |

Reviewer: ```suggestion
| edit a recipe        | `edit <name of recipe> [-n <new name of recipe> -d <new prep time of recipe> -ss <new serving size/portions of recipe> -i <new ingredients separated by commas> -s <new steps separated by commas> -t <new tags separated by commas>]` <br/> or <br/> `edit <recipe number from list> [-n <new name of recipe> -d <new prep time of recipe> -ss <new serving size/portions of recipe> -i <new ingredients separated by commas> -s <new steps separated by commas> -t <new tags separated by commas>]` |
```

| delete a recipe      | `delete <recipe name>` <br> or <br> `delete -x <recipe number from list>`                                                                                                                                      |
| find a recipe        | `find <keyword> [other keywords separated by commas]`                                                                                                                                                          |
| list a recipe        | `list`                                                                                                                                                                                                         |

> | view specific recipe | `view <recipe number from list>` <br/> or <br/> `view <recipe name>`                                                                                                                                           |

Reviewer: ```suggestion
| view specific recipe | `view <recipe name>` <br/> or <br/> `view <recipe number from list>`                                                                                                                                           |
```
